src/tests/Microsoft.PowerFx.Interpreter.Tests/SymbolValueTests.cs

```diff
@@ -618,6 +618,56 @@ public async Task TestMismatch2()
             Assert.Throws<InvalidOperationException>(() => symValues.Set(slot, FormulaValue.New("abc")));
         }
 
+        // Demonstrate how to use ThisItem in a loop.
+        // All Loop iterations can access common global state,
+        // but also have their own loop-specific identifer. 
+
+        [Fact]
+        public async Task ThisItemTest()
+        {
+            var symTableGlobals = new SymbolTable { DebugName = "Globals" };
+            var slotGlobal = symTableGlobals.AddVariable("globalvar", FormulaType.Number);
+
+            // Note from PowerFx perspective, there is nothing special about the identifier 'ThisItem'.
+            // It's all determined by SymbolTable topology. 
+            var symTableThisItem = new SymbolTable { DebugName = "ThisItem" };
+            var slotThisItem = symTableThisItem.AddVariable("ThisItem", FormulaType.Number);
+
+            // Combine together
+            var symTable = ReadOnlySymbolTable.Compose(symTableThisItem, symTableGlobals);
+
+            var expr = "globalvar + ThisItem";
+            
+            // Compile once outside the loop, then run many times. 
+            var engine = new RecalcEngine();
+            var check = engine.Check(expr, symbolTable: symTable);
+            check.ThrowOnErrors();
+
+            // Now move to runtime...
+            var run = check.GetEvaluator();
+
+            // all loop bodies can read-only access common global state. 
+            var symValuesGlobals = symTableGlobals.CreateValues();
+            symValuesGlobals.Set(slotGlobal, FormulaValue.New(1000));
+
+            // Loop body can be run in parallel since each eval is independent. 
+            // An there are no mutations on shared state.
+            Parallel.For(0, 10, (i) =>
+            {
+                // Each loop body gets its own iteration-specific values.
+                var symValuesThisItem = symTableThisItem.CreateValues();
+                symValuesThisItem.Set(slotThisItem, FormulaValue.New(i));
+
+                var symValues = symTable.CreateValues(symValuesGlobals, symValuesThisItem);
```

Comment:
>symValuesGlobals [](http://example.com/codeflow?start=54&length=16)

Is symValuesGlobals immutable? If it is, would be possible to add a new test case showing what would it happen if something tries to change it in a lopp?